packages/babel-parser/test/helpers/runFixtureTests.js

```diff
@@ -107,19 +109,33 @@ export function runThrowTestsWithEstree(fixturesPath, parseFunction) {
 }
 
 function save(test, ast) {
-  // Ensure that RegExp and Errors are serialized as strings
-  forceToString(RegExp, () =>
-    forceToString(Error, () =>
-      fs.writeFileSync(test.expect.loc, JSON.stringify(ast, null, "  ")),
-    ),
+  forceToString(() =>
+    fs.writeFileSync(test.expect.loc, JSON.stringify(ast, null, 2)),
   );
 }
 
-function forceToString(obj, cb) {
-  const { toJSON } = obj.prototype;
-  obj.prototype.toJSON = obj.prototype.toString;
+// Ensure that RegExp, BigInt, and Errors are serialized as strings
+function forceToString(cb) {
+  const originalToJSONMap = new Map();
+  const notJSONparseableObj = [RegExp, Error];
+
+  if (typeof BigInt !== "undefined") {
+    notJSONparseableObj.push(BigInt);
+  }
+
+  for (const obj of notJSONparseableObj) {
+    const { toJSON } = obj.prototype;
+    originalToJSONMap.set(obj, toJSON);
+    obj.prototype.toJSON = function() {
+      return this.toString();
```

Comment:
We need to ensure we're not passing parameters through here because `BigInt.prototype.toString()` takes a radix argument that must be between `[2-36]` and throws if it gets called with a different argument. It has a default value, though, so it's fine to not call it with anything.